history/event.go

```diff
@@ -207,18 +209,43 @@ func (e *Event) UnmarshalJSON(in []byte) error {
 		if err := json.Unmarshal(wireEvent.MetadataBytes, &metadata); err != nil {
 			return err
 		}
-		e.Metadata = metadata
+		e.Metadata = &metadata
 		break
 	default:
 		if len(wireEvent.MetadataBytes) > 0 {
-			var metadata interface{}
+			var metadata UnknownEventMetadata
 			if err := json.Unmarshal(wireEvent.MetadataBytes, &metadata); err != nil {
 				return err
 			}
-			e.Metadata = metadata
+			e.Metadata = &metadata
```

Comment:
Does this need to be a pointer? (it's a bit unusual to be passing around pointers to maps, unless you really want someone to swap the whole thing)